File: 48-rotate-image/48-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        // Transpose The matrix
        int n = size(matrix);
        int m = size(matrix[0]);
        if (n == 1) return;
        for (int i = 0;i < n;i++) {
            for (int j = i;j < n;j++) {
                swap(matrix[i][j],matrix[j][i]);
            }
        }
       // 2-pointer approch
        for (int i = 0;i < n;i++) {
            reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};

File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, mat: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        #Transepose
        ROW,COL = len(mat),len(mat[0])
        for i in range(ROW):
            for j in range(i+1,COL):
                #swap
                mat[i][j],mat[j][i] = mat[j][i],mat[i][j]
            mat[i] = mat[i][::-1]